---
title: "Visualize Epi Conservation Data"
author: "ks"
date: "2024-01-03"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(tidyverse)
```


Load CpG level data on pwds.
```{r pwd}
data_dir <- c("~kims/Google Drive/My Drive/shibata/Shibata EPIC Data/datajy/data")
load(file.path(data_dir,"ec.rdat"))
load(file.path(data_dir,"crcdev.rda"))
load(file.path(data_dir,"cpgannot.rda"))
```

Combine with annotation
```{r merge-ec-cpgannot}
annoec <- merge(x=cpgannot, 
                y=ec,
                all.x = FALSE,
                all.y = TRUE) %>%
           arrange(chr,mapinfo)
```


There are different numbers of missing PWD measures in tumors M & I, but it is sparse.
```{r fraction-missing-data}
mean(is.na(annoec$Mpwd))
mean(is.na(annoec$Ipwd))
```

Global distribution of PWD.
```{r distn-cpg-pwd}
summary(annoec$Mpwd)
summary(annoec$Ipwd)
```

What fraction of features have PWD < 0.05? (the cutoff in the paper)
```{r}
mean(annoec$Mpwd < 0.05, na.rm=T)
mean(annoec$Ipwd < 0.05, na.rm=T)
```

Summarize by gene, merge with sc-RNAseq deviances & plot. First, how does the distribution of gene average PWD compare to the distribution of CpG PWDs?  

```{r gene-ec}
gene_ec <- annoec %>% group_by(ucsc_refgene_name) %>%
    summarise(
       Mpwd_avg  = mean(Mpwd, na.rm=TRUE),
       Ma_avg    = mean(Ma, na.rm=TRUE),
       Mpwd_3q   = quantile(Mpwd, probs=0.75, na.rm=T ),
       Ipwd_avg  = mean(Ipwd, na.rm=TRUE),
       Ia_avg    = mean(Ia, na.rm=TRUE),
       Ipwd_3q   = quantile(Ipwd, probs=0.75, na.rm=T ),
       n = n()
    )  %>% filter(n > 3) 
dim(gene_ec)
gene_ec <- gene_ec %>%
          rename(name = ucsc_refgene_name)
```

```{r}
gene_dev_ec <- 
    merge(x = gene_ec,
          y = crcdev)
dim(gene_dev_ec)
```


Global distribution of gene-level average PWD.
```{r distn-gene-pwd_avg}
summary(gene_dev_ec$Mpwd_avg)
summary(gene_dev_ec$Ipwd_avg)
```

What fraction of genes have average PWD < 0.05? (the cutoff in the paper)
```{r}
mean(gene_dev_ec$Mpwd_avg < 0.05, na.rm=T)
mean(gene_dev_ec$Ipwd_avg < 0.05, na.rm=T)
```
 

Count number of CpGs without the DNA methylation data. Are they the same? Yes.
```{r gene-ec-tI}
gene_ecI <- annoec %>% group_by(ucsc_refgene_name) %>%
    summarise(
       nI = n()
    )  %>% filter(nI > 3) 

gene_ecI <- gene_ecI %>%
          rename(name = ucsc_refgene_name)

gene_dev_ecI <- 
    merge(x = gene_ecI,
          y = crcdev)

gene_dev_ec_MI <- 
    merge(x = gene_ec,
          y = gene_ecI)
summary(gene_dev_ec_MI$n - gene_dev_ec_MI$nI)
```



```{r fig4-tumorM, fig.height=2, fig.width=3, echo=FALSE}
rvar <- cor(gene_dev_ec$Mpwd_avg,
            gene_dev_ec$dev) %>%
          round(2)

dat_text <- data.frame(
  label = paste("cor =",rvar)
)

pe <- ggplot(gene_dev_ec,aes(x=Mpwd_avg, y=jdev)) +
  geom_point(shape = ".", alpha = 0.4) + 
  geom_smooth(method="lm",color="red") +
  theme( axis.text=element_text(size=8),
         axis.title=element_text(size=10),
         panel.background = element_blank(),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA)) +
  ylab("Expression variability
       (log2 Deviance)") +
  xlab("conservation (PWD)") +
  ylim(-0.5,20) + xlim(0,0.25) + ggtitle("Tumor M") +
  geom_text( 
      data    = dat_text,
      mapping = aes(x = 0.15, y = 12, label = label),
      hjust   = -0.05,
      vjust   = -1
    )

pe 

ggsave("figs/sesame-tumorM.png",    plot = last_plot(),    width = 3,    height = 2,    dpi = 300,    units = "in" )
```


And for tumor I:
```{r fig4-tumorI, fig.height=2, fig.width=3, echo=FALSE}
rvar <- cor(gene_dev_ec$Ipwd_avg,
            gene_dev_ec$dev,use="complete.obs") %>%
          round(2)

dat_text <- data.frame(
  label = paste("cor =",rvar)
)

pe <- ggplot(gene_dev_ec,aes(x=Ipwd_avg, y=jdev)) +
  geom_point(shape = ".", alpha = 0.4) + 
  geom_smooth(method="lm",color="red") +
  theme( axis.text=element_text(size=8),
         axis.title=element_text(size=10),
         panel.background = element_blank(),
        panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.border = element_rect(colour = "black", fill=NA)) +
  ylab("Expression variability
       (log2 Deviance)") +
  xlab("conservation (PWD)") +
  ylim(-0.5,20) + xlim(0,0.25) + ggtitle("Tumor I") +
  geom_text( 
      data    = dat_text,
      mapping = aes(x = 0.15, y = 12, label = label),
      hjust   = -0.05,
      vjust   = -1
    )

pe 

#ggsave("figs/sesame-tumorI.png",    plot = last_plot(),    width = 3,    height = 2,    dpi = 300,    units = "in" )
```

Let's see if number of CpGs predicts PWD. Then, let's create 2 groups of expression variability. Those with none and those with some. Then let's see if number of CpGs predicts PWD in each group. I think a lot of our PWDs = 0 are due to low numbers of CpGs, and this occurs more frequently in the genes that lack variability in expression (deviance = 0).

```{r}
#gene_dev_ec
```


```{r}
#annoec %>% filter(ucsc_refgene_name=="A2M")
```

Lastly, restrict # CpGs to >10 and watch the correlations get stronger.

```{r sI}
sessionInfo()
```

